include/mfmg/dealii/lanczos.templates.hpp

```diff
@@ -430,6 +365,20 @@ void Lanczos<OperatorType, VectorType>::details_calc_evecs(
   }
 }
 
+// Initial guess is a linear combination of a
+// constant vector (to try to capture "smooth" eigenmodes of PDE problems)
+// and a random vector based on different random seeds.
+// ISSUE: should a different initial guess strategy be used?
+template <typename OperatorType, typename VectorType>
+void Lanczos<OperatorType, VectorType>::details_set_initial_guess(
+    VectorType &initial_guess, int seed) const
+{
+  std::mt19937 gen(seed);
+  std::uniform_real_distribution<double> dist(0, 1);
+  std::generate(initial_guess.begin(), initial_guess.end(),
+                [&]() { return 1 + dist(gen); });
```

Comment:
I didn't catch that before.  Just do `std::uniform_real_distribution<double> dist(1, 2);` and `{ return dist(gen); }`

Reply:
No, this way the intention is more clear.

Reply:
You can't be serious.  Saying you want a random real number between 0 and 1 and then shifting it by one is more clear than generating it between 1 and 2??

Reply:
![1 slwikdf7w3-unnnuuxkulw](https://user-images.githubusercontent.com/296978/52726469-3ea81400-2f81-11e9-8bb6-186cc532822b.png)